oteapi/plugins/entry_points.py

```diff
@@ -0,0 +1,505 @@
+"""Load plugins through entry points.
+
+This module deals with handling all plugged in strategies through the entry points API
+and importlib metadata API.
+
+Special functionality is put in place to handle the OTE-API-specific entry points.
+
+Since the entry points are information complete in and of themselves, there is no need
+to import actual strategy classes until they are truly needed.
+This therefore implements lazy loading of all plugin strategies.
+"""
+import importlib
+import re
+from collections import abc
+from enum import Enum
+from functools import lru_cache
+from importlib.metadata import entry_points as get_entry_points
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:  # pragma: no cover
+    from importlib.metadata import EntryPoint
+    from typing import Any, Iterable, Iterator, Optional, Set, Tuple, Type, Union
+
+    from oteapi.interfaces import IStrategy
+
+
+class EntryPointNotFound(Exception):
+    """A given strategy implementation (class) cannot be found from the given entry
+    point value."""
+
+
+class StrategyType(Enum):
+    """An enumeration of available strategy types.
+
+    Available strategy types:
+
+    - download
+    - filter
+    - mapping
+    - parse
+    - resource
+    - transformation
+
+    """
+
+    DOWNLOAD = "download"
+    FILTER = "filter"
+    MAPPING = "mapping"
+    PARSE = "parse"
+    RESOURCE = "resource"
+    TRANSFORMATION = "transformation"
+
+    @lru_cache
+    def map_to_field(self) -> str:
+        """Map enumeration value to the strategy type's field."""
+        return {
+            "download": "scheme",
+            "filter": "filterType",
+            "mapping": "mappingType",
+            "parse": "mediaType",
+            "resource": "accessService",
+            "transformation": "transformation_type",
+        }[self.value]
+
+    @classmethod
+    @lru_cache
```

Comment:
True - I'm trying to start learning to use it and thought it was nice? But perhaps not?

Reply:
Caching make sense when the underlying calculation is expensive. A dict lookup is cheap (actually cheaper than the cache since in include additional logics). Hence, a cache make no sense in this case.